lib/src/over_react_test/react_util.dart

```diff
@@ -481,8 +481,36 @@ Element queryByTestId(dynamic root, String value, {String key = defaultTestIdKey
 ///     </div>
 ///
 ///     queryAllByTestId(renderedInstance, 'value'); // returns both `inner` `<div>`s
-List<Element> queryAllByTestId(dynamic root, String value, {String key = defaultTestIdKey}) {
-  return findDomNode(root).querySelectorAll('[$key~="$value"]');
+List<Element> queryAllByTestId(dynamic root, String value, {String key = defaultTestIdKey, bool searchInShadowDom = false, int shadowDepth = 1}) {
+  return _findAllDeep(findDomNode(root), _makeTestIdSelector(value, key: key), searchInShadowDom: searchInShadowDom, findMany: true);
+}
+
+String _makeTestIdSelector(String value, {String key = defaultTestIdKey}) => '[$key~="$value"]';
+
+dynamic /* Element | List<Element> */ _findAllDeep(Element root, String itemSelector, {bool searchInShadowDom = false, bool findMany = true, int depth = 1}) {
+  List<Element> nodes = [];
+  var currentDepth = 0;
+  depth ??= 1;
+  var lightElement = root.querySelector(itemSelector);
+  if (!findMany && lightElement != null){
+    return lightElement;
+  }
+  void recursiveSeek(dynamic _root) {
+    nodes = [...nodes, ..._root.querySelectorAll(itemSelector)];
+    if (!findMany && nodes.isNotEmpty) {
+      return;
+    }
+    // This is kinda gross but I figured this was only ever used in test situations so its probably alright.
+    if (depth != null && currentDepth < depth && searchInShadowDom == true) {
+      var foundShadows = _root.querySelectorAll('*').where((el) => el.shadowRoot != null).map((el) => el.shadowRoot).toList();
```

Comment:
I'm just now learning about shadow dom while reviewing this, what's the reason you don't just keep going until there's no more `foundShadows` instead of taking in a `depth` arg? Will it go on infinitely?

Reply:
it's highly unlikely that it would go on infinitely, the depth part was mainly to like provide a way to rein it in a bit in case there are many shadowroots within shadow roots. Honestly it could probably be removed i just was thinking it might be nice in some niche situations and most of the time it probably won't need to go deeper than 1 or 2.